### PR TITLE
tests: subsys: settings: functional: nvs & fcb split for CI

### PR DESCRIPTION
**Pull request by kapi-no**
Splitted configuration for NVS & FCB so it is possible for the CI to pick it up.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

### CODE REVIEW

**Comment by kapi-no** [comment]
NVS test is failing as expected now (so it is seen by CI):

https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/49680/4/console

I will try to fix it in the next update

**Comment by kapi-no** [comment]
Test are passing now: https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/49688. Fixing gitlint

**Review by nvlsianpu — APPROVED** [review]
looks good, works fine

**Comment by nvlsianpu** [comment]
@andrewboie If you can look at qemu_x86 changes  it would be great.

**Comment by nvlsianpu** [comment]
@ioannisg can you review?

**Review by carlescufi — APPROVED** [review]

File: tests/subsys/settings/functional/nvs/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+project(NONE)
+
+FILE(GLOB app_sources ../src/*.c)
+target_sources(app PRIVATE ${app_sources})
```
Comment (marc-hb) [inline tests/subsys/settings/functional/nvs/CMakeLists.txt]: @kapi-no" the ".." parent directory above triggers CMake issue https://gitlab.kitware.com/cmake/cmake/issues/19475 and causes a reproducibility regression: it makes the build directory very long and depending on the source directory. See a similar example and more details (including ASCII art) at fix https://github.com/zephyrproject-rtos/hal_nordic/pull/6

Now unlike hal_nordic/pull/6, I bet there are other ways to fix this that do not necessarily require a new zephyr_library(). For instance these source files could simply be added from a higher level `CMakeLists.txt` which would be more consistent with normal CMake practices.

Can people familiar with this part of the code tell what is the proper way to organize it? Thanks.

**Pull request merged**